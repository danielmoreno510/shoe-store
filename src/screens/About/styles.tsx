import {StyleSheet} from 'react-native';
import {heightPercentageToDP, widthPercentageToDP} from '../../utils/responsiveUtils';

export default StyleSheet.create({
  container: {
    flex: 1,
    padding: widthPercentageToDP('5%'),
    backgroundColor: 'white',
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
  },
  subtitle: {
    fontSize: 24,
  },
  titleContainer: {
    padding: 4,
    paddingHorizontal: 12,
  },
  itemsContainer: {
    padding: 4,
    paddingHorizontal: 18,
  },
  item: {
    fontSize: 20,
  },
  text: {
    paddingTop: 8,
    fontSize: 16,
    textAlign: 'justify',
  },
  image: {
    marginTop: 20,
    width: widthPercentageToDP('90%'),
    height: heightPercentageToDP(`${widthPercentageToDP('100%') / 40}`),
  },
});
