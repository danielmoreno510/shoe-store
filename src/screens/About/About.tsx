import React from 'react';
import {ScrollView, View, Text, Image} from 'react-native';

import styles from './styles';
import coverage from '../../assets/images/coverage.png';
import appcenter from '../../assets/images/appcenter.png';
import pipeline from '../../assets/images/pipeline.png';

const About = () => (
  <ScrollView style={styles.container}>
    <Text style={styles.title}>Sneaker City</Text>
    <View style={styles.titleContainer}>
      <Text style={styles.subtitle}>Technologies</Text>
      <View style={styles.itemsContainer}>
        <Text style={styles.item}>- React Native</Text>
        <Text style={styles.item}>- TypeScript</Text>
        <Text style={styles.item}>- Redux - Saga</Text>
        <Text style={styles.item}>- esLint</Text>
        <Text style={styles.item}>- Test (Jest)</Text>
      </View>
    </View>
    <View style={styles.titleContainer}>
      <Text style={styles.subtitle}>Tools</Text>
      <View style={styles.itemsContainer}>
        <Text style={styles.item}>- Reactotron</Text>
        <Text style={styles.item}>- Xcode</Text>
        <Text style={styles.item}>- Prettier</Text>
      </View>
    </View>
    <Text style={styles.text}>This project has a coverage of 81%</Text>
    <Image source={coverage} style={styles.image} />
    <Text style={styles.text}>
      In other projects I have implemented continuous integration processes with implementation within
      appcenter to download the ios and android builds
    </Text>
    <Image source={pipeline} style={styles.image} />
    <Image source={appcenter} style={styles.image} />
  </ScrollView>
);

export default About;
