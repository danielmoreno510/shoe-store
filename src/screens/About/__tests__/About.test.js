import React from 'react';
import {shallow} from 'enzyme';

import About from '../About';

describe('About component', () => {
  describe('GIVEN a set of valid props', () => {
    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<About />);

      it('THEN should display a regular About', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });
  });
});
