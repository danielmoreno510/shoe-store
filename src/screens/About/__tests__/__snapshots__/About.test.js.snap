// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular About 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 37.5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    Sneaker City
  </Text>
  <View
    style={
      Object {
        "padding": 4,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Technologies
    </Text>
    <View
      style={
        Object {
          "padding": 4,
          "paddingHorizontal": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        - React Native
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        - TypeScript
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        - Redux - Saga
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        - esLint
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        - Test (Jest)
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Tools
    </Text>
    <View
      style={
        Object {
          "padding": 4,
          "paddingHorizontal": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        - Reactotron
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        - Xcode
      </Text>
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        - Prettier
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingTop": 8,
        "textAlign": "justify",
      }
    }
  >
    This project has a coverage of 81%
  </Text>
  <Image
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Object {
        "height": 250,
        "marginTop": 20,
        "width": 675,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "paddingTop": 8,
        "textAlign": "justify",
      }
    }
  >
    In other projects I have implemented continuous integration processes with implementation within appcenter to download the ios and android builds
  </Text>
  <Image
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Object {
        "height": 250,
        "marginTop": 20,
        "width": 675,
      }
    }
  />
  <Image
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Object {
        "height": 250,
        "marginTop": 20,
        "width": 675,
      }
    }
  />
</ScrollView>
`;
